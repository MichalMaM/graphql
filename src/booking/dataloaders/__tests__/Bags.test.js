// @flow

import createBagsDataLoader from '../Bags';

jest.mock('../../../common/services/JsonFetcher', () => ({
  fetchJson: () => {
    const fakeData = {
      baggage: [{ bag: { x: 'foo' } }, { bag: { x: 'bar' } }],
      pending_baggage: [{ bag: { x: 'baz' } }],
    };
    return Promise.resolve(fakeData);
  },
}));

const BagsDataLoader = createBagsDataLoader('');

it('returns baggage AND pending baggage', async () => {
  await expect(BagsDataLoader.load(3243242)).resolves.toMatchSnapshot();
});
