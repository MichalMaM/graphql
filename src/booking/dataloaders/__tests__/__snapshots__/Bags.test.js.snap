// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns baggage AND pending baggage 1`] = `
Array [
  Object {
    "isPending": false,
    "x": "foo",
  },
  Object {
    "isPending": false,
    "x": "bar",
  },
  Object {
    "isPending": true,
    "x": "baz",
  },
]
`;
